controllers/mysqlcluster_controller_test.go

```diff
@@ -76,44 +112,537 @@ spec:
+			cluster.Status.ServerIDBase = &serverIDBase
+			cluster.Spec.VolumeClaimTemplates = []mocov1alpha1.PersistentVolumeClaim{
 				{
-					Name:       "mysql",
-					Protocol:   corev1.ProtocolTCP,
-					Port:       moco.MySQLPort,
-					TargetPort: intstr.FromInt(moco.MySQLPort),
+					ObjectMeta: mocov1alpha1.ObjectMeta{
+						Name: "test-volume",
+					},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceStorage: *resource.NewQuantity(1<<10, resource.BinarySI),
+							},
+						},
+					},
 				},
+			}
+
+			isUpdated, err := reconciler.createOrUpdateStatefulSet(ctx, reconciler.Log, cluster)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(isUpdated).Should(BeTrue())
+
+			sts := &appsv1.StatefulSet{}
+			err = k8sClient.Get(ctx, client.ObjectKey{Name: moco.UniqueName(cluster), Namespace: cluster.Namespace}, sts)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			var testClaim *corev1.PersistentVolumeClaim
+			var dataClaim *corev1.PersistentVolumeClaim
+			for i, v := range sts.Spec.VolumeClaimTemplates {
+				if v.Name == "test-volume" {
+					testClaim = &sts.Spec.VolumeClaimTemplates[i]
+				}
+				if v.Name == mysqlDataVolumeName {
+					dataClaim = &sts.Spec.VolumeClaimTemplates[i]
+				}
+			}
+			Expect(testClaim).ShouldNot(BeNil())
+			Expect(dataClaim).ShouldNot(BeNil())
+
+			isUpdated, err = reconciler.createOrUpdateStatefulSet(ctx, reconciler.Log, cluster)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(isUpdated).Should(BeFalse())
+		})
+
+		It("should return error, when volumeTemplate contains mysql-data", func() {
+			oldSts := &appsv1.StatefulSet{}
+			err := k8sClient.Get(ctx, client.ObjectKey{Name: moco.UniqueName(cluster), Namespace: cluster.Namespace}, oldSts)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = k8sClient.Delete(ctx, oldSts)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			serverIDBase := mathrand.Uint32()
+			cluster.Status.ServerIDBase = &serverIDBase
+			cluster.Spec.VolumeClaimTemplates = []mocov1alpha1.PersistentVolumeClaim{
 				{
-					Name:       "mysqlx",
-					Protocol:   corev1.ProtocolTCP,
-					Port:       moco.MySQLXPort,
-					TargetPort: intstr.FromInt(moco.MySQLXPort),
+					ObjectMeta: mocov1alpha1.ObjectMeta{
+						Name: mysqlDataVolumeName,
+					},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceStorage: *resource.NewQuantity(1<<10, resource.BinarySI),
+							},
+						},
+					},
 				},
 			}
 
+			_, err = reconciler.createOrUpdateStatefulSet(ctx, reconciler.Log, cluster)
+			Expect(err).Should(HaveOccurred())
+		})
+
+	})
+
+	Context("CronJob", func() {
+		It("should create cron job", func() {
```

Comment:
Some additional tests are needed, such as the CronJob has the correct command with the agent token.